### PR TITLE
Issue #151 update paths in dockerfile to fix codespaces build

### LINKED ISSUE #151: Fix docker build failure for codespaces image
## Description

The docker build for codespaces is failing because of a missing file. The cause if that the build starts from the parent directory.

The build is triggered like

```bash
docker build . -t codespaces-dotnet -f .devcontainer/Dockerfile
```

 But then the dockerfile looks for `library-scripts/...` which does not exist in the current directory. It is under `.devcontainer/`

Fix the build.

### PR DESCRIPTION
**Pull request by AAkindele**
# Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR

Update the file copy commands in the Dockerfile to use the path from the project root. That is where the build starts.
Tested this in a forked repo and was able to get a successful build. Was also able to run through the Readme instructions in GitHub codespaces to get the app up and running. Was also able to run the app in codespaces and open the forwarded port in the browser.

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #151

### CODE REVIEW

**Review by sivamu — APPROVED** [review]

**Review by sivamu — APPROVED** [review]
Tried this out and changes work well.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,10 +1,21 @@
 {
   "name": "helium-csharp",
-  "image": "retaildevcrew/codespaces-dotnet",
```
Comment (sivamu) [inline .devcontainer/devcontainer.json:3]: I'm assuming that without specifying the image or the build, it's using the upstream vscode image?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -16,27 +16,24 @@ ARG  NODE_VERSION="none"
...
+# To use the Docker CLI inside a container, use the bind mount option to add the docker socket to the container.
+# Example: "docker run -v '/var/run/docker.sock:/var/run/docker.sock' ..."
 ARG  INSTALL_DOCKER="true"
-COPY library-scripts/docker-debian.sh /tmp/library-scripts/
+COPY .devcontainer/library-scripts/docker-debian.sh /tmp/library-scripts/
 RUN  if [ "${INSTALL_DOCKER}" = "true" ]; then \
-        bash /tmp/library-scripts/docker-debian.sh "true" "vscode"; \
-     else \
-        echo '#!/bin/bash\nexec "$@"' > /usr/local/share/docker-init.sh && chmod +x /usr/local/share/docker-init.sh; \
+        bash /tmp/library-scripts/docker-debian.sh $USERNAME; \
```
Comment (sivamu) [inline .devcontainer/Dockerfile]: can put this in a single line similar to the azure and node setups

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (AAkindele) [reply .devcontainer/devcontainer.json:3]: correct. i also tested that docker commands still work with the upstream image without the "mount" option.

File: .devcontainer/library-scripts/docker-debian.sh [hunk 3]
```diff
@@ -18,4 +44,7 @@ LATEST_COMPOSE_VERSION=$(curl -sSL "https://api.github.com/repos/docker/compose/
 curl -sSL "https://github.com/docker/compose/releases/download/${LATEST_COMPOSE_VERSION}/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
 chmod +x /usr/local/bin/docker-compose
 
+groupadd -g 800 docker
+usermod -a -G docker $USERNAME
```
Comment (bartr) [inline .devcontainer/library-scripts/docker-debian.sh]: you don't need to do this if username=root

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -16,27 +16,22 @@ ARG  NODE_VERSION="none"
 ENV  NVM_DIR=/usr/local/share/nvm
 ENV  NVM_SYMLINK_CURRENT=true \
      PATH=${NVM_DIR}/current/bin:${PATH}
-COPY library-scripts/node-debian.sh /tmp/library-scripts/
+COPY .devcontainer/library-scripts/node-debian.sh /tmp/library-scripts/
```
Comment (bartr) [inline .devcontainer/Dockerfile]: should we just copy all of the scripts at once?  each copy creates a layer and the files are small

Comment (AAkindele) [reply .devcontainer/Dockerfile]: i think that is a good idea for our case because we're going to use all the files in that directory. i'll update it.

**Pull request merged**